Tests arch arm irq vector table fix for armv6

ioannisg opened the pull request:
Increasing coverage for ARMv6 - adding the IRQ Vector Table test for Cortex-M Baseline, too.

ioannisg committed:
tests: arch: arm_irq_vector_table: enable test for ARMv6-M CPUs

This commit enables running the test in CI for Cortex-M Baseline
architecture. As STIR register is not present in Baseline cores,
we need a minor modification in the test code so it uses NVIC
intrisics instead of setting the STIR for ARMv6-M.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
tests: arch: arm_irq_vector_table: enable test for nRF51x platforms

This commit enables running the test in CI for nRF51-based
platforms, which require special IRQ lines for the IRQs
used in the test. An additional customization is needed
for QEMU_CORTEX_M0, which uses a different system timer
based on TIMER0 peripheral.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

anangl commented on tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c:
```diff
@@ -16,8 +16,8 @@
-#if defined(CONFIG_SOC_SERIES_NRF52X)
-/* The customized solution for nRF52X-based platforms
+#if defined(CONFIG_SOC_SERIES_NRF51X) || defined(CONFIG_SOC_SERIES_NRF52X)
+/* The customized solution for nRF5X-based platforms
```
Same correction needed in [line 26](https://github.com/zephyrproject-rtos/zephyr/pull/19856/files#diff-c64ef6ac0266b8f4ff2c414cd2733cdeR26) (can't add a comment there).

ioannisg replied:
Thank, will add

anangl commented on tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c:
```diff
@@ -152,24 +166,34 @@ void test_arm_irq_vector_table(void)
+void timer0_nrf_isr(void);
+vth __irq_vector_table _irq_vector_table[RTC1_IRQn + 1] = {
+	nrf_power_clock_isr,
+	0, 0, 0, 0, 0, 0, 0, timer0_nrf_isr, isr0, isr1, isr2, 0, 0, 0, 0, 0,
```
These trailing `0`s don't seem to be useful.

ioannisg replied:
Thanks, will fix

anangl commented on tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c:
```diff
@@ -134,7 +147,8 @@ void test_arm_irq_vector_table(void)
+#if defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE) || \
+	defined(CONFIG_SOC_TI_LM3S6965_QEMU)
 		/* the QEMU does not simulate the
 		 * STIR register: this is a workaround
```
Shouldn't this comment be updated as well?

ioannisg replied:
No, ARMv6-M is well known not to have a STIR - I don't think this is a needed explanation. The TI SOC is ARMv7 but just the QEMU is incompletely emulating it, so we've added the comment.

anangl commented on tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c:
```diff
@@ -152,24 +166,33 @@ void test_arm_irq_vector_table(void)
+#if defined(CONFIG_SOC_SERIES_NRF51X) || defined(CONFIG_SOC_SERIES_NRF52X)
+#if defined(CONFIG_BOARD_QEMU_CORTEX_M0)
+void timer0_nrf_isr(void);
+vth __irq_vector_table _irq_vector_table[TIMER0_IRQn + 1 + 3] = {
```
Why do we actually specify sizes of these arrays? Compilers should be able to determine those by counting the provided entries.

ioannisg replied:
Good question ;)
Totally missed that during all the reworks I've done in this test. I 'll add an extra commit to fix this

ioannisg committed:
tests: arch: arm_irq_vector_table: no explicit const array size

We do not need to explicitly define the size of the const
array elements for the customized IRQ vector, used in the
test. The compiler should be able to infer the actual size
by counting the provided entries.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg reviewed (commented):
@anangl added a commit that removes the explicit array size definitions on the const irq vector tables 
thanks!
